rasa/shared/core/generator.py

```diff
@@ -159,23 +163,31 @@ def init_copy(self) -> "TrackerWithCachedStates":
             self.is_rule_tracker,
         )
 
+    @contextmanager
+    def _skip_states_manager(self) -> bool:
+        self._skip_states = True
```

Comment:
I'd be in favor of protect this variable even further and use a double underscore